src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -0,0 +1,320 @@
+            }
+
+            if (!reportDiagnostic)
+            {
+                if (symbol is IMethodSymbol || symbol is IPropertySymbol)
+                {
+                    reportDiagnostic = PropertyOrMethodSymbolHasAttribute(symbol, requiresPreviewFeaturesSymbols, requiresPreviewFeaturesSymbolsToUsageType, runtimeFeatureType);
+                }
+            }
+
+            if (reportDiagnostic)
+            {
+                if (requiresPreviewFeaturesSymbolsToUsageType.TryGetValue(symbol, out PreviewFeatureUsageType usageType))
+                {
+                    if (usageType == PreviewFeatureUsageType.StaticAbstract)
+                    {
+                        context.ReportDiagnostic(symbol.CreateDiagnostic(StaticAbstractIsPreviewFeatureRule));
+                    }
+                    else if (usageType == PreviewFeatureUsageType.EmptyInterface)
+                    {
+                        context.ReportDiagnostic(symbol.CreateDiagnostic(ImplementsEmptyPreviewInterfaceRule, symbol.Name));
+                    }
+                }
+                else
+                {
+                    context.ReportDiagnostic(symbol.CreateDiagnostic(GeneralPreviewFeatureAttributeRule, symbol.Name));
+                }
+            }
+        }
+
+        private static void BuildSymbolInformationFromOperations(OperationAnalysisContext context, ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols)
+        {
+            IOperation operation = context.Operation;
+            if (OperationUsesPreviewFeatures(operation, requiresPreviewFeaturesSymbols, out ISymbol? symbol))
+            {
+                if (operation is ICatchClauseOperation catchClauseOperation)
+                {
+                    operation = catchClauseOperation.ExceptionDeclarationOrExpression;
+                }
+
+                context.ReportDiagnostic(operation.CreateDiagnostic(GeneralPreviewFeatureAttributeRule, symbol!.Name));
+            }
+        }
+
+        private static bool OperationUsesPreviewFeatures(IOperation operation, ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols, out ISymbol? symbol)
+        {
+            symbol = GetOperationSymbol(operation);
+            // CHange the operation here to ExceptionDeclarationOrExpression
+            if (symbol == null)
+            {
+                return false;
+            }
+
+            return OperationUsesPreviewFeatures(symbol, requiresPreviewFeaturesSymbols);
+        }
+
+        private static ISymbol? GetOperationSymbol(IOperation operation)
+            => operation switch
+            {
+                IInvocationOperation iOperation => iOperation.TargetMethod,
+                IObjectCreationOperation cOperation => cOperation.Constructor,
+                IPropertyReferenceOperation pOperation => pOperation.Property,
```

Comment:
Are parameter reference and local reference operations skipped intentionally?

Reply:
Yup. My reasoning was:
If a preview type is passed in as a parameter (or is a local variable or is a type instance), the use of any method/property/field on that parameter/variable/type will be marked by the analyzer. Or the variable remains unused. I can't think of a common scenario where we miss tagging preview feature usage.